### ISSUE TITLE
Accessibility: ARIA violations, missing labels and semantic structure issues on homepage

### ISSUE THREAD

**Issue by shivambisen** [cilium/cilium.io#700]
### Story

I was going through the website and spotted some accessibility issues on the homepage of the Cilium.io website. These should be addressed to improve accessibility for screen readers and crawlers.

#### 1. [aria-hidden="true"] elements contain focusable descendents
- Focusable elements inside `[aria-hidden="true"]`  it means there are some hidden components still have <a> tags or buttons hidden form screen readers.
![Image](https://github.com/user-attachments/assets/ff9b65a2-f7c8-4a54-aaa7-ecad1258650b)

#### 2. Missing `[lang]` attribute on `<html>`
- The `<html>` tag does not declare a `lang` attribute, which can hinder proper pronunciation for screen readers.
 ![Image](https://github.com/user-attachments/assets/79eebcd5-57af-4910-9444-18f448b7360d)

#### 3. Lists not using `<li>` properly
- Unordered lists (`<ul>`) contain elements that are not defined inside (`<li>`) valid list children.
![Image](https://github.com/user-attachments/assets/f5b4bd2c-9c42-4093-9c8d-6dffe983f1cb)

---
